lib/bolt/pal.rb

```diff
@@ -291,17 +291,35 @@ def get_task_info(task_name)
     end
 
     def list_plans
+      require 'puppet-strings'
+      require 'puppet-strings/yard'
       in_bolt_compiler do |compiler|
         errors = []
         plans = compiler.list_plans(nil, errors).map { |plan| [plan.name] }.sort
         errors.each do |error|
           @logger.warn(error.details['original_error'])
         end
-        plans
+        plans.reject { |plan| private_plan?(plan.first) }
       end
     end
 
-    def get_plan_info(plan_name)
+    def private_plan?(plan_name)
+      pp_path = plan_location(plan_name)[:path]
+      if File.exist?(pp_path)
+        PuppetStrings::Yard.setup!
+        YARD::Logger.instance.level = :error
+        YARD.parse(pp_path)
+
+        plan = YARD::Registry.at("puppet_plans::#{plan_name}")
+
+        private_plan = plan.tag(:private) && plan.tag(:private).text == 'true'
```

Comment:
It might make sense to just add this to `get_plan_info` rather than make a bespoke method for it.

Reply:
Thats a good point. I guess by the time we have already done the Yard parsing, avoiding a little bit more data munging is not saving much energy.